client/src/views/SharePoint.tsx

```diff
@@ -56,15 +63,22 @@ export const SharePoint = () => {
         ],
     }
 
-    const [topSitesByPageView, setTopSitesByPageView] = useState<Array<{ siteName: string; pageViewCount: number }>>([])
+
+    const topSitesByPageView = useMemo(() => {
+
+        const sortedData = sitesActivity.filter((dt: SiteActivityWithSites) => dt.pageViewCount !== undefined)
+        sortedData.sort((a1, a2) => a2.pageViewCount - a1.pageViewCount)
```

Comment:
This is failing the build 
      
       TS18048: 'a2.pageViewCount' is possibly 'undefined'.
           68 |
           69 |         const sortedData = sitesActivity.filter((dt: SiteActivityWithSites) => dt.pageViewCount !== undefined)
         > 70 |         sortedData.sort((a1, a2) => a2.pageViewCount - a1.pageViewCount)
              |                                     ^^^^^^^^^^^^^^^^